metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/sql/DirectSqlTable.java

```diff
@@ -0,0 +1,217 @@
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.sql.Types;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This class makes direct sql calls to get/set table metadata.
+ *
+ * @author amajumdar
+ * @since 1.2.0
+ */
+@Slf4j
+@Transactional("hiveTxManager")
+public class DirectSqlTable {
+    private static final String PARAM_METADATA_LOCK = "metadata_lock";
+    private final Registry registry;
+    private final JdbcTemplate jdbcTemplate;
+    private final HiveConnectorFastServiceMetric fastServiceMetric;
+    private final String catalogName;
+
+    /**
+     * Constructor.
+     *
+     * @param connectorContext     server context
+     * @param jdbcTemplate         JDBC template
+     * @param fastServiceMetric    fast service metric
+     */
+    public DirectSqlTable(
+        final ConnectorContext connectorContext,
+        final JdbcTemplate jdbcTemplate,
+        final HiveConnectorFastServiceMetric fastServiceMetric
+    ) {
+        this.catalogName = connectorContext.getCatalogName();
+        this.registry = connectorContext.getRegistry();
+        this.jdbcTemplate = jdbcTemplate;
+        this.fastServiceMetric = fastServiceMetric;
+    }
+
+    /**
+     * Returns true if table exists with the given name.
+     * @param name table name
+     * @return true if table exists with the given name.
+     */
+    @Transactional(readOnly = true)
+    public boolean exists(final QualifiedName name) {
+        final long start = registry.clock().wallTime();
+        boolean result = false;
+        try {
+            final Object qResult = jdbcTemplate.queryForObject(SQL.EXIST_TABLE_BY_NAME,
+                new String[]{name.getDatabaseName(), name.getTableName()},
+                new int[]{Types.VARCHAR, Types.VARCHAR}, Integer.class);
+            if (qResult != null) {
+                result = true;
+            }
+        } catch (EmptyResultDataAccessException e) {
+            return false;
+        } finally {
+            this.fastServiceMetric.recordTimer(
+                HiveMetrics.TagTableExists.getMetricName(), registry.clock().wallTime() - start);
+        }
+        return result;
+    }
+
+    /**
+     * Returns all the table names referring to the given <code>uris</code>.
+     *
+     * @param uris         locations
+     * @param prefixSearch if true, we look for tables whose location starts with the given <code>uri</code>
+     * @return map of uri to list of partition names
+     * @throws UnsupportedOperationException If the connector doesn't implement this method
+     */
+    @Transactional(readOnly = true)
+    public Map<String, List<QualifiedName>> getTableNames(final List<String> uris, final boolean prefixSearch) {
+        final long start = registry.clock().wallTime();
+        // Create the sql
+        final StringBuilder queryBuilder = new StringBuilder(SQL.GET_TABLE_NAMES_BY_URI);
```

Comment:
For testing you might want to think about breaking these SQL building blocks into their own method so you can quickly check them for bugs.